utils/context.go

```diff
@@ -0,0 +1,31 @@
+package utils
+
+import (
+	"context"
+
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/metadata"
+)
+
+const MetadataKey = "viam-metadata"
+
+func ContextWithMetadata(ctx context.Context) (context.Context, map[string][]string) {
+	md := make(map[string][]string)
+	ctx = context.WithValue(ctx, MetadataKey, md)
+	return ctx, md
+}
+
+func ContextWithMetadataUnaryClientInterceptor(ctx context.Context, method string, req, reply interface{}, cc *grpc.ClientConn, invoker grpc.UnaryInvoker, opts ...grpc.CallOption) error {
```

Comment:
let's add some unit tests around this as well as some tests in camera for testing it out!

Reply:
Done, I added a test showing the replaypcd camera sets the gRPC metadata on calls to NextPointCloud, and another showing the interceptor correctly finds the gRPC metadata in the response header and injects it into the context when it's a context with metadata. I ended up setting up an inject camera for that second test so I'm now thinking it may be better to merge them into one; let me know what you think.